### PR TITLE
Allow to specify schema location by namespace

### PR DESCRIPTION
**Pull request by cebe**
there is an undocumented configuration option "known_locations" which
allows replacing remote files with local variants.

this adds a new option to specify schemaLocation for a namespace
`known_namespace_locations`.

Depends on https://github.com/goetas-webservices/xsd-reader/pull/45 so
you need to update composer constraints when merging this.

### CODE REVIEW

**Comment by goetas** [comment]
> Depends on goetas-webservices/xsd-reader#45 so you need to update composer constraints when merging this.

I have added the composer constraints on your branch

**Comment by goetas** [comment]
Hmm, it looks that this will oblige me to drop php 5.x support, as the v0.3.5 is php 7 only

**Comment by cebe** [comment]
I can backport the patch to the 0.2.x branch

**Comment by goetas** [comment]
> I can backport the patch to the 0.2.x branch

that would be great! (of course if it is not too much effort).
I have still some projects on php 5.6 using this lib.

(but sooner or later php 5.6 support will have to be dropped :smiley: )

File: composer.json [hunk 1]
```diff
@@ -23,7 +23,7 @@
     "symfony/dependency-injection": "^2.2|^3.0|^4.0",
     "symfony/yaml": "^2.1|^3.0|^4.0",
     "symfony/config": "^2.2|^3.0|^4.0",
-    "goetas-webservices/xsd-reader": "^0.2|^0.3.1",
+    "goetas-webservices/xsd-reader": "^0.3.5",
```
Comment (goetas) [inline composer.json]: now we can have `^0.2.7|^0.3.5` here

Comment (cebe) [reply composer.json]:
```suggestion
    "goetas-webservices/xsd-reader": "^0.2.7|^0.3.5",
```

**Comment by cebe** [comment]
ready to merge.

**Comment by goetas** [comment]
Thanks!

**Comment by goetas** [comment]
Tagged as v0.3.5

**Pull request merged**